pilot/pkg/networking/core/v1alpha3/listener_inbound.go

```diff
@@ -156,55 +154,33 @@ type ServiceInstancePort struct {
 }
 
 func (lb *ListenerBuilder) buildInboundHBONEListeners() []*listener.Listener {
-	vhost := &route.VirtualHost{
-		Name:    "inbound-hbone-connect",
-		Domains: []string{"*"},
-	}
 	inboundChainConfigs := lb.buildInboundChainConfigs()
-	for _, cc := range inboundChainConfigs {
-		// TODO passthrough
-		p := strconv.Itoa(int(cc.port.TargetPort))
-		destination := "inbound-hbone" + "|" + p
-		vhost.Routes = append(vhost.Routes, &route.Route{
-			Match: &route.RouteMatch{
-				PathSpecifier: &route.RouteMatch_ConnectMatcher_{ConnectMatcher: &route.RouteMatch_ConnectMatcher{}},
-				Headers: []*route.HeaderMatcher{
-					istiomatcher.HeaderMatcher(":authority", "*:"+p),
-				},
-			},
-			Action: &route.Route_Route{Route: &route.RouteAction{
-				UpgradeConfigs: []*route.RouteAction_UpgradeConfig{{
-					UpgradeType:   "CONNECT",
-					ConnectConfig: &route.RouteAction_UpgradeConfig_ConnectConfig{},
-				}},
-
-				ClusterSpecifier: &route.RouteAction_Cluster{Cluster: destination},
-			}},
-			TypedPerFilterConfig: map[string]*anypb.Any{
-				xdsfilters.ConnectAuthorityFilter.Name: xdsfilters.ConnectAuthorityEnabledSidecar,
-			},
-		})
-	}
-	l := &listener.Listener{
-		Name:    "inbound-hbone",
-		Address: util.BuildAddress("0.0.0.0", model.HBoneInboundListenPort),
-		FilterChains: []*listener.FilterChain{
-			{
-				TransportSocket: buildDownstreamTLSTransportSocket(lb.authnBuilder.ForHBONE().TCP),
-				Filters: []*listener.Filter{
-					xdsfilters.IstioNetworkAuthenticationFilterShared,
-					buildHBONEConnectionManager(vhost),
-				},
-			},
+	routes := []*route.Route{{
+		Match: &route.RouteMatch{
+			PathSpecifier: &route.RouteMatch_ConnectMatcher_{ConnectMatcher: &route.RouteMatch_ConnectMatcher{}},
 		},
-	}
-
-	accessLogBuilder.setListenerAccessLog(lb.push, lb.node, l, istionetworking.ListenerClassSidecarInbound)
+		Action: &route.Route_Route{Route: &route.RouteAction{
+			UpgradeConfigs: []*route.RouteAction_UpgradeConfig{{
+				UpgradeType:   "CONNECT",
+				ConnectConfig: &route.RouteAction_UpgradeConfig_ConnectConfig{},
+			}},
 
-	var listeners []*listener.Listener
-	listeners = append(listeners, l)
+			ClusterSpecifier: &route.RouteAction_Cluster{Cluster: InternalName},
+		}},
+		TypedPerFilterConfig: map[string]*anypb.Any{
+			// Note the difference with the waypoint below.
+			xdsfilters.ConnectAuthorityFilter.Name: xdsfilters.ConnectAuthorityEnabledSidecar,
```

Comment:
What is the diff with `waypoint` connect_terminate listener routes, hard to see with bare eyes, if no diff please share one

Reply:
Added comment.